client/me/concierge/reschedule/calendar-step.js

```diff
@@ -72,17 +90,41 @@ class CalendarStep extends Component {
 					scheduleId={ WPCOM_CONCIERGE_SCHEDULE_ID }
 				/>
 
-				<AvailableTimePicker
-					actionText={ translate( 'Reschedule to this date' ) }
-					availableTimes={ availableTimes }
-					currentUserLocale={ currentUserLocale }
-					disabled={ signupForm.status === CONCIERGE_STATUS_BOOKING || ! appointmentDetails }
-					description={ translate( 'Please select a day to reschedule your Concierge session.' ) }
-					onBack={ null }
-					onSubmit={ this.onSubmit }
-					site={ site }
-					signupForm={ signupForm }
-				/>
+				<CompactCard>
+					{ translate(
+						'Please select the desired timezone and day to reschedule your Concierge session.'
+					) }
+				</CompactCard>
+
+				{ appointmentDetails && (
+					<div>
+						<CompactCard>
+							<FormFieldset>
+								<FormLabel>{ translate( "What's your timezone?" ) }</FormLabel>
+								<Timezone
+									includeManualOffsets={ false }
+									name="timezone"
+									onSelect={ this.setTimezone }
+									selectedZone={ appointmentDetails.meta.timezone }
+								/>
+								<FormSettingExplanation>
+									{ translate( 'Choose a city in your timezone.' ) }
+								</FormSettingExplanation>
+							</FormFieldset>
```

Comment:
The lines 102 ~ 103 look almost identical to [the lines 68-79 in book/info-step.js](https://github.com/Automattic/wp-calypso/blob/master/client/me/concierge/book/info-step.js#L68-L79). Would it be possible to extract it into an individual component?

Reply:
Not quite sure if we need a new component just for a form field.

Reply:
Good call @unDemian — I don't think this is complicated enough to merit abstraction.

Reply:
Cool. I'm also good with keeping it this way. :)